pkg/hub/addon/addon.go

```diff
@@ -1,66 +1,230 @@
 package addon
 
 import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/open-cluster-management/cluster-proxy-addon/pkg/config"
+	"github.com/open-cluster-management/cluster-proxy-addon/pkg/helpers"
+	bindata "github.com/open-cluster-management/cluster-proxy-addon/pkg/hub/addon/bindata"
+	"github.com/openshift/library-go/pkg/assets"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2"
 	"open-cluster-management.io/addon-framework/pkg/agent"
-	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	"open-cluster-management.io/api/addon/v1alpha1"
 	clusterv1 "open-cluster-management.io/api/cluster/v1"
 
 	"github.com/openshift/library-go/pkg/operator/events"
 
 	certificatesv1 "k8s.io/api/certificates/v1"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	genericScheme = runtime.NewScheme()
+	genericCodecs = serializer.NewCodecFactory(genericScheme)
+	genericCodec  = genericCodecs.UniversalDeserializer()
 )
 
 const (
-	addOnName = "cluster-proxy"
-	agentName = "cluster-proxy-agent"
+	addOnGroup         = "system:open-cluster-management:addon:cluster-proxy"
+	agentUserName      = "system:open-cluster-management:cluster:%s:addon:cluster-proxy:agent:cluster-proxy-addon-agent"
+	clusterAddOnGroup  = "system:open-cluster-management:cluster:%s:addon:cluster-proxy"
+	authenticatedGroup = "system:authenticated"
 )
 
+func init() {
+	scheme.AddToScheme(genericScheme)
+}
+
 type clusterProxyAddOnAgent struct {
-	kubeClient kubernetes.Interface
-	recorder   events.Recorder
-	agentImage string
+	recorder      events.Recorder
+	hubKubeClient *kubernetes.Clientset
```

Comment:
using `kubernetes.Interface`, it will be easy to test

Reply:
Done